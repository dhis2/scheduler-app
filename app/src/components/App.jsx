import 'typeface-roboto';
import 'material-design-icons-iconfont/dist/material-design-icons.css';
import React from 'react';
import { Provider } from 'react-redux';
import HeaderBar from '@dhis2/ui/widgets/HeaderBar';
import i18n from '@dhis2/d2-i18n';
import '../styles/override.css';
import store from '../store';
import MessagePanel from './MessagePanel';
import ContentLoader from './ContentLoader';
import AddD2Context from './AddD2Context';

const App = ({ d2 }) => (
    <Provider store={store}>
        <AddD2Context d2={d2}>
            <div>
                <HeaderBar appName={i18n.t('Scheduler')} />
                <MessagePanel />
                <ContentLoader d2={d2} />
            </div>
        </AddD2Context>
    </Provider>
);

export default App;
