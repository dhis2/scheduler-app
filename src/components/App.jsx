import 'typeface-roboto';
import 'material-design-icons-iconfont/dist/material-design-icons.css';

import React from 'react';
import { Provider } from 'react-redux';
import i18n from '@dhis2/d2-i18n';
import { HeaderBar } from '@dhis2/ui-widgets';
import { CssReset } from '@dhis2/ui-core';
import { Provider as RuntimeProvider } from '@dhis2/app-runtime';
import store from '../store';
import MessagePanel from './MessagePanel';
import ContentLoader from './ContentLoader';
import AddD2Context from './AddD2Context';
import '../styles/override.css';

const App = ({ d2, baseUrl, apiVersion }) => (
    <RuntimeProvider config={{
        baseUrl,
        apiVersion,
    }}
    >
        <Provider store={store}>
            <AddD2Context d2={d2}>
                <div>
                    <CssReset />
                    <HeaderBar appName={i18n.t('Scheduler')} />
                    <MessagePanel />
                    <ContentLoader d2={d2} />
                </div>
            </AddD2Context>
        </Provider>
    </RuntimeProvider>
);

export default App;
