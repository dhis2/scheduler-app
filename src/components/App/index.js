/* eslint-disable import/no-unused-modules */

export { default } from './App.jsx'
