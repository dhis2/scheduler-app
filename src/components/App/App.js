import React from 'react'
import { CssVariables } from '@dhis2/ui'
import { Routes } from '../Routes'
import { AuthWall } from '../AuthWall'
import { Store } from '../Store'
import { PageWrapper } from '../PageWrapper'
import './App.css'

// eslint-disable-next-line import/no-unassigned-import
import '../../locales'
// The above is necessary for translations to work

const App = () => (
    <React.Fragment>
        <CssVariables spacers colors theme />
        <PageWrapper>
            <AuthWall>
                <Store>
                    <Routes />
                </Store>
            </AuthWall>
        </PageWrapper>
    </React.Fragment>
)

export default App
