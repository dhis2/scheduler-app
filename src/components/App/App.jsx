import React from 'react'
import { CssVariables } from '@dhis2/ui'
import { Routes } from '../Routes'
import { AuthWall } from '../AuthWall'
import { Store } from '../Store'
import { PageWrapper } from '../PageWrapper'
import './App.css'

/* eslint-disable-next-line import/no-unassigned-import -- Necessary for translations to work */
import '../../locales'

const App = () => (
    <React.Fragment>
        <CssVariables spacers colors theme />
        <PageWrapper>
            <AuthWall>
                <Store>
                    <Routes />
                </Store>
            </AuthWall>
        </PageWrapper>
    </React.Fragment>
)

export default App
