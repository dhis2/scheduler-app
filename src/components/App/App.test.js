import React from 'react'
import { shallow } from 'enzyme'
import App from './App'

jest.mock('../../hooks/locale', () => ({
    useLocale: jest.fn(),
}))

describe('<App>', () => {
    it('renders without errors', () => {
        shallow(<App />)
    })
})
