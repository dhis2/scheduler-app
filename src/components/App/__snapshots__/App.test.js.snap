// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> renders correctly 1`] = `
<Fragment>
  <CssVariables
    colors={false}
    elevations={false}
    layers={false}
    spacers={true}
    theme={false}
  />
  <PageWrapper>
    <AuthWall>
      <Routes />
    </AuthWall>
  </PageWrapper>
</Fragment>
`;
